hello.rb

```diff
@@ -0,0 +1,36 @@
+require 'data_mapper'
+require './app/data_mapper_setup'
+require 'bcrypt'
+
+class User
+
+  include DataMapper::Resource
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  validates_confirmation_of :password
+  validates_uniqueness_of   :email
+
+  property :id, Serial
+  property :email, String, required: true
+  property :name, String
+  property :username, String
+  property :password_digest, Text
+  property :password_confirmation, Text
+
+  def password=(password)
+    @password = password
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  def self.authenticate(username, password)
```

Comment:
Lint/UnusedMethodArgument: Unused method argument - `username`. If it's necessary, use `_` or `_username` as an argument name to indicate that it won't be used.